monitoring/monitoring/utils.py

```diff
@@ -0,0 +1,165 @@
+import numpy as np
+import pandas as pd
+import plotly.express as px
+
+from database.connection import GreenDB
+from database.tables import SustainabilityLabelsTable
+
+from . import CONNECTION_FOR_TABLE
+
+green_db = GreenDB()
+
+
+def dates() -> tuple:
+    """
+    Gets latest timestamp for green-db and sustainability-labels tables and converts it to date
+    format.
+
+    Returns:
+        Tuple: with date of last extraction from green-db and date of last update from
+        sustainability-labels
+    """
+    last_extraction = green_db.get_latest_timestamp().date()
+    last_label_update = green_db.get_latest_timestamp(SustainabilityLabelsTable).date()
+    return last_extraction, last_label_update
+
+
+def latest_extraction_summary() -> tuple:
```

Comment:
Consider other return type, maybe `dict` which would greatly increase the readability of the app code because currently it is hard to know whats in the returned values.

Reply:
I renamed functions and changed return types [2976091](https://github.com/calgo-lab/green-db/pull/91/commits/29760919033122cb8a9e5423f449a373dcddb833)
Merge state
Add more commits by pushing to the [monitoring](https://github.com/calgo-lab/green-db/tree/monitoring) branch on [calgo-lab/green-db](https://github.com/calgo-lab/green-db).

Changes requested
1 review requesting changes by reviewers with write access. 
1 change requested
Some checks haven’t completed yet
2 in progress and 4 successful checks
@github-actions
Test / pytest-extract (pull_request) In progress — This check has started...
[Details](https://github.com/calgo-lab/green-db/runs/8160508864?check_suite_focus=true)
@github-actions
Lint / black (pull_request) Successful in 6s
[Details](https://github.com/calgo-lab/green-db/runs/8160508721?check_suite_focus=true)
@github-actions
Test / pytest-scraping (pull_request) In progress — This check has started...
[Details](https://github.com/calgo-lab/green-db/runs/8160508725?check_suite_focus=true)
@github-actions
Lint / isort (pull_request) Successful in 5s
[Details](https://github.com/calgo-lab/green-db/runs/8160509011?check_suite_focus=true)
@github-actions
Lint / flake8 (pull_request) Successful in 7s
[Details](https://github.com/calgo-lab/green-db/runs/8160508857?check_suite_focus=true)
@github-actions
Lint / mypy (pull_request) Successful in 11s
[Details](https://github.com/calgo-lab/green-db/runs/8160509110?check_suite_focus=true)
Merging is blocked
Merging can be performed automatically once the requested changes are addressed.
 You can also [open this in GitHub Desktop](https://desktop.github.com/) or view .
@AdriaSG

 
Add heading textAdd bold text, <Cmd+b>Add italic text, <Cmd+i>
Add a quote, <Cmd+Shift+.>Add code, <Cmd+e>Add a link, <Cmd+k>
Add a bulleted list, <Cmd+Shift+8>Add a numbered list, <Cmd+Shift+7>Add a task list, <Cmd+Shift+l>
Directly mention a user or team
Reference an issue, pull request, or discussion
Add saved reply
Leave a comment
No file chosen
Attach files by dragging & dropping, selecting or pasting them.
Styling with Markdown is supported
Remember, contributions to this repository should follow our [GitHub Community Guidelines](https://docs.github.com/articles/github-community-guidelines).
 ProTip! Add [.patch](https://github.com/calgo-lab/green-db/pull/91.patch) or [.diff](https://github.com/calgo-lab/green-db/pull/91.diff) to the end of URLs for Git’s plaintext views.
Reviewers
@[se-jaeger](https://github.com/se-jaeger)
se-jaeger
Requested changes must be addressed to merge this pull request.

Still in progress? 
Assignees
No one—
Labels
None yet
Projects
None yet
Milestone
No milestone
Development
Successfully merging this pull request may close these issues.

None yet

Notifications
Customize
You’re receiving notifications because you authored the thread.
2 participants
@AdriaSG
@se-jaeger